lib/doh.c

```diff
@@ -69,7 +69,14 @@ static const char * const errors[]={
 
 static const char *doh_strerror(DOHcode code)
 {
+#if defined(__GNUC__) && \
+  ((__GNUC__ < 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ <= 4)))
+  /* Workaround for:
+     warning: comparison of unsigned expression >= 0 is always true */
```

Comment:
is there a reason we need that warning, maybe we can just turn it off? i don't think it provides a lot of value.

Reply:
I'm not be in favour of losing a warning globally for all versions
because of a single outlier / bug. In this case this is a very old gcc.

It'd be possible (and probably nicer) to turn it off via cmake/autotools
for this specific build case, but we'd need to know which `-W` option
this warning is. In clang and newer gcc versions it's called
`-Wtautological-compare`, but that's not recognized by 4.4.

Reply:
Ok. It looks like it's `-Wno-type-limits` for gcc 4.4 but if you think it's needed I have no objections

Reply:
Nice, I added `-Wno-type-limits` globally, it makes this local workaround unnecessary, thanks!